MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java

```diff
@@ -0,0 +1,390 @@
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.style.expressions.Expression;
+import com.mapbox.mapboxsdk.style.layers.HeatmapLayer;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+
+import java.io.InputStream;
+
+import static com.mapbox.mapboxsdk.style.expressions.Expression.heatmapDensity;
+import static com.mapbox.mapboxsdk.style.expressions.Expression.interpolate;
+import static com.mapbox.mapboxsdk.style.expressions.Expression.linear;
+import static com.mapbox.mapboxsdk.style.expressions.Expression.literal;
+import static com.mapbox.mapboxsdk.style.expressions.Expression.rgb;
+import static com.mapbox.mapboxsdk.style.expressions.Expression.rgba;
+import static com.mapbox.mapboxsdk.style.expressions.Expression.zoom;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.heatmapColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.heatmapIntensity;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.heatmapOpacity;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.heatmapRadius;
+
+public class MultipleHeatmapStylingActivity extends AppCompatActivity
+  implements OnMapReadyCallback {
+
+  private static final String HEATMAP_SOURCE_ID = "HEATMAP_SOURCE_ID";
+  private static final String HEATMAP_LAYER_ID = "HEATMAP_LAYER_ID";
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private Expression[] listOfHeatmapColors;
+  private Expression[] listOfHeatmapRadiusStops;
+  private Float[] listOfHeatmapIntensityStops;
+  private int index;
+  private boolean firstLoad;
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    index = 0;
+    firstLoad = true;
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+    setContentView(R.layout.activity_multiple_heatmap_styling);
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    MultipleHeatmapStylingActivity.this.mapboxMap = mapboxMap;
+    CameraPosition cameraPositionForFragmentMap = new CameraPosition.Builder()
+      .target(new LatLng(34.056684, -118.254002))
+      .zoom(11.047)
+      .build();
+    mapboxMap.animateCamera(
+      CameraUpdateFactory.newCameraPosition(cameraPositionForFragmentMap), 2600);
+    addHeatmapDataSource();
+    initHeatmapColors();
+    initHeatmapRadiusStops();
+    initHeatmapIntensityStops();
+    addHeatmapLayer();
+    findViewById(R.id.switch_heatmap_style_fab).setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        if (index == listOfHeatmapColors.length - 1) {
+          index = 0;
+        }
+        if (firstLoad) {
```

Comment:
why not remove this boolean and move the `index++` to the top of this method?

Reply:
both done